src/transaction/boot_sr.c

```diff
@@ -1577,70 +1578,74 @@ xboot_initialize_server (const BOOT_CLIENT_CREDENTIAL * client_credential, BOOT_
+  error_code = cfg_read_directory (&dir, true);
+  if (error_code != NO_ERROR)
+    {
+      goto exit_on_error;
+    }
+#endif // WINDOWS && DONT_USE_MANDATORY_LOCK_IN_WINDOWS
 
-  if (dir != NULL && ((db = cfg_find_db_list (dir, client_credential->get_db_name ())) != NULL))
+  if (dir != NULL)
     {
-      if (db_overwrite == false)
+      db = cfg_find_db_list (dir, client_credential->get_db_name ());
+      if (db != NULL)
 	{
-	  /* There is a database with the same name and we cannot overwrite it */
-	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_BO_DATABASE_EXISTS, 1, client_credential->get_db_name ());
-	  goto exit_on_error;
-	}
-      else
-	{
-	  /*
-	   * Delete the database.. to make sure that all backups, log archives, and
-	   * so on are removed... then continue...
-	   *
-	   * Note: we do not call xboot_delete since it shuttdown the system and
-	   *       update database.txt that we have a read copy of its content.
-	   */
+	  if (db_overwrite == false)
+	    {
+	      /* There is a database with the same name and we cannot overwrite it */
+	      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_BO_DATABASE_EXISTS, 1, client_credential->get_db_name ());
+	      goto exit_on_error;
+	    }
+	  else
+	    {
+	      /*
+	       * Delete the database.. to make sure that all backups, log archives, and
+	       * so on are removed... then continue...
+	       *
+	       * Note: we do not call xboot_delete since it shutdown the system and
+	       *       update database.txt that we have a read copy of its content.
+	       */
 
-	  /* Note: for database replacement, we need to remove the old database with its original path! */
-	  memset (original_namebuf, 0, sizeof (original_namebuf));
+	      /* Note: for database replacement, we need to remove the old database with its original path! */
+	      memset (original_namebuf, 0, sizeof (original_namebuf));
 
-	  /* Compose the original full name of the database */
-	  snprintf (original_namebuf, sizeof (original_namebuf), "%s%c%s", db->pathname, PATH_SEPARATOR, db->name);
+	      /* Compose the original full name of the database */
+	      snprintf (original_namebuf, sizeof (original_namebuf), "%s%c%s", db->pathname, PATH_SEPARATOR, db->name);
 
-	  error_code = boot_remove_all_volumes (thread_p, original_namebuf, db->logpath, log_prefix, false, true);
-	  if (error_code != NO_ERROR)
-	    {
-	      goto exit_on_error;
+	      error_code = boot_remove_all_volumes (thread_p, original_namebuf, db->logpath, log_prefix, false, true);
+	      if (error_code != NO_ERROR)
+		{
+		  goto exit_on_error;
+		}
 	    }
 	}
-    }
 
-  if (dbtxt_vdes != NULL_VOLDES)
-    {
-      fileio_dismount (thread_p, dbtxt_vdes);	/* unlock the directory file */
-      dbtxt_vdes = NULL_VOLDES;
+      if (dbtxt_vdes != NULL_VOLDES)
+	{
+	  fileio_dismount (thread_p, dbtxt_vdes);	/* unlock the directory file */
+	  dbtxt_vdes = NULL_VOLDES;
+	}
       cfg_free_directory (dir);
       dir = NULL;
+      db = NULL;
```

Comment:
replace with [optional] assert:
```suggestion
      assert ((db_overwrite && db == nullptr) || !db_overwrite);
```
or even better
```suggestion
      assert (db == nullptr);
```
and `db = nullptr;` should be moved to else scope ending at ~1638.
I understand that if `db` is non-null here, there's a logic problem.

Reply:
db may be non-null if db_overwrite was true. db is set to null as an indication that it is not used below.